indico/modules/receipts/client/js/printing/TemplateParameterEditor.jsx

```diff
@@ -68,6 +82,19 @@ function CustomField({name, value, type, attributes, validations, onChange}) {
         required={validations.required}
       />
     );
+  } else if (type === 'image') {
+    return (
+      <EventImageSelectField
+        label={attributes.label}
+        name={name}
+        value={value}
+        required={validations.required}
+        loading={loadingImages}
+        onChange={onChange}
```

Comment:
I'm not sure if this is getting called once when you open the dropdown or if it's caused by something else, but currently we send both the GET request for the image choices and also a POST request to render a preview immediately when opening the dropdown.

Reply:
this happens because `useEffect` (in the `Previewer`) doesnt do deep comparison of objects, so it thinks it the form values are being updated every time the form is re-rendered...
I was checking for solutions but none of them are pretty... - https://copyprogramming.com/howto/react-useeffect-comparing-objects

Reply:
won't this whole problem go away when you switch to not reloading it every time you open the dropdown? :)

Reply:
done, but it still happens when you click the refresh button...